### ISSUE TITLE
text-generation-inference docker builds are not reproducible due to missing Cargo.lock causing builds to fail on previous versions

### ISSUE THREAD

**Issue by charlesmelby** [huggingface/optimum-neuron#677]
### System Info

```shell
system: inf2.48xlarge instance
OS: Amazon Linux 2023

build was done with v0.0.23 but mainline has same issue
```

### Who can help?

@dacorvo 

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction (minimal, reproducible, runnable)

To reproduce:
```
git clone https://github.com/huggingface/optimum-neuron --branch=v0.0.23
cd optimum-neuron
make neuronx-tgi
```

Error message:
```
error: package `bitstream-io v2.5.0` cannot be built because it requires rustc 1.79 or newer, while the currently active rustc version is 1.75.0
Either upgrade to rustc 1.79 or newer, or use
cargo update bitstream-io@2.5.0 --precise ver
where `ver` is the latest version of `bitstream-io` supporting rustc 1.75.0
thread 'main' panicked at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/cargo-chef-0.1.62/src/recipe.rs:204:27:
Exited with status code: 101
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

### Expected behavior

I expect the build to succeed.
Cargo.lock has bitstream-io v2.2.0 but it's not being used so invalid dependencies are being collected.

Build goes through when Cargo.lock is added as in TGI repo
https://github.com/huggingface/text-generation-inference/blob/8094ecfc9ef22c838fa7d49db4af8301539619e3/Dockerfile#L8

**Comment by charlesmelby** [comment]
@dacorvo 
Sorry didn't tag you properly before.

**Comment by schauaib** [comment]
Heelo,
Any update about this ?

**Comment by dacorvo** [comment]
This was fixed as part of #696
